server/src/main/java/org/elasticsearch/cluster/BatchedTaskExecutor.java

```diff
@@ -56,14 +58,15 @@ public final void clusterStatePublished(ClusterState newClusterState) {
     public void clusterStatePublished() {}
 
     @Override
-    public final ClusterState execute(BatchExecutionContext<T> batchExecutionContext) throws Exception {
+    public final ClusterState execute(BatchExecutionContext<Task> batchExecutionContext) throws Exception {
         var initState = batchExecutionContext.initialState();
         var clusterState = initState;
         for (final var taskContext : batchExecutionContext.taskContexts()) {
             try (var ignored = taskContext.captureResponseHeaders()) {
                 var task = taskContext.getTask();
-                clusterState = executeTask(task, clusterState);
-                taskContext.success(() -> taskSucceeded(task));
+                Tuple<ClusterState, TaskResult> result = executeTask(task, clusterState);
+                clusterState = result.v1();
+                taskContext.success(() -> taskSucceeded(task, result.v2()));
```

Comment:
Probably not a huge problem but might be worth avoiding capturing the intermediate state here:

```suggestion
                final var taskResult = result.v2();
                taskContext.success(() -> taskSucceeded(task, taskResult));
```